modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -799,8 +798,7 @@ object CurrencySnapshotAcceptanceManager {
             unexpired.foldLeft(acc.getOrElse(address, Balance.empty)) { (currentBalance, allowSpend) =>
```

Comment:
I think it is still error prone. The logic of applying `unexpired` and `expired` is splitted into 2 separate `foldLeft` calls. So it can be that applying unexpired allow spend on address succeeds but applying expired one fails. Even if we make it in a single foldLeft then it still won't be transactional 🤔 .

The question is if it should be transactional per address so all expried+unexpired allow spends for address should be applied or none.